xain/grpc/participant.py

```diff
@@ -33,6 +35,53 @@ def rendezvous(channel):
         response = reply.response
 
 
+def startTraining(channel):
+    stub = coordinator_pb2_grpc.CoordinatorStub(channel)
+    req = coordinator_pb2.StartTrainingRequest()
+    # send request to start training
+    reply = stub.StartTraining(req)
+    print(f"Participant received: {type(reply)}")
+    theta, epochs, epoch_base = reply.theta, reply.epochs, reply.epoch_base
+    print(f"epochs: {epochs}, epoch_base: {epoch_base}")
+    # iterate over theta, deserialising and logging...
+    print(f"theta of type {type(theta)}:")
+    for proto_nda in theta:
+        nda = proto_to_ndarray(proto_nda)
+        print(f"{nda}")
+    # now simulate some training...
+
+
+def endTraining(channel):
+    stub = coordinator_pb2_grpc.CoordinatorStub(channel)
+    # build request starting with theta update
+    nda1, nda2 = np.arange(20, 30), np.arange(30, 40)
+    num = 2
+    print(f"Participant requests: theta' {nda1}, {nda2}; {num} examples")
+    theta_prime = [ndarray_to_proto(nda1), ndarray_to_proto(nda2)]
+    theta_update = coordinator_pb2.EndTrainingRequest.ThetaUpdate(
+        theta_prime=theta_prime, num_examples=num
+    )
+    # history data
+    k1, k2 = "aaa", "bbb"
+    v1, v2 = [1.1, 2.1], [3.1, 4.1]
```

Comment:
Instead of having some hardcoded values here could we instead turn this into a test.

This would also help us make the `start` and `end` training methods more generic